File: Programmers/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int red) {
    vector<int> answer;
    
    vector<vector<int>> redSize;
    vector<int> redCandidate;
    for(int i = 1; i < red + 1; i++){
        if(red % i == 0){
            if(i > red/i)
                break;
            redCandidate.push_back(red / i);
            redCandidate.push_back(i);
            redSize.push_back(redCandidate);
        }
        redCandidate.clear();
    }
    
    int cntCandidates = redSize.size();
    for(int i = 0; i < cntCandidates; i++){
        int x = redSize[i][0] + 2;
        int y = redSize[i][1] + 2;
        if(x * y == brown + red){
            answer.push_back(x);
            answer.push_back(y);
            break;
        }
    }
    return answer;
}

File: Programmers/카펫.swift
import Foundation

func solution(_ brown: Int, _ yellow: Int) -> [Int] {
    var x = 0
    var y = 0
    for i in stride(from: Int(sqrt(Double(yellow))), to: 0, by: -1) {
        x += 1
        if yellow % x == 0 {
            y = yellow / x
            if (x + 2) * (y + 2) - yellow == brown { break }
        }
    }
    return [y + 2, x + 2]
}
